### PR TITLE
8271063: Print injected fields for InstanceKlass

### PR DESCRIPTION
**Pull request by coleenp**
I added code to print the injected fields in InstanceKlass.  I also removed field offset sorting for do_nonstatic_fields because it's not necessary when used for purposes other than printing the fields.  I also added a gtest.
Tested with tier1 all Oracle platforms, tier2-3 on linux-x64-debug, windows-x64-debug.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8271063](https://bugs.openjdk.java.net/browse/JDK-8271063): Print injected fields for InstanceKlass

### Reviewers
 * [Frederic Parain](https://openjdk.java.net/census#fparain) (@fparain - Committer) ⚠️ Review applies to 7f8e7421440baf3223194486e8d6a91931f81c30
 * [Harold Seigel](https://openjdk.java.net/census#hseigel) (@hseigel - **Reviewer**) ⚠️ Review applies to 7f8e7421440baf3223194486e8d6a91931f81c30
 * [Yi Yang](https://openjdk.java.net/census#yyang) (@kelthuzadx - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4858/head:pull/4858` \
`$ git checkout pull/4858`

Update a local copy of the PR: \
`$ git checkout pull/4858` \
`$ git pull https://git.openjdk.java.net/jdk pull/4858/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4858`

View PR using the GUI difftool: \
`$ git pr show -t 4858`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4858.diff">https://git.openjdk.java.net/jdk/pull/4858.diff</a>

</details>

### CODE REVIEW

**Review by fparain — APPROVED** [review]
Looks good to me.

Fred

**Comment by coleenp** [comment]
Thanks Fred!

**Review by hseigel — APPROVED** [review]
Changes look good!
Thanks, Harold

File: src/hotspot/share/oops/instanceKlass.cpp [hunk 1]
```diff
@@ -1623,43 +1623,63 @@ void InstanceKlass::do_local_static_fields(void f(fieldDescriptor*, Handle, TRAP
...
-      fields_sorted[j + 1] = i;
-      j += 2;
+      cl->do_field(&fd);
     }
   }
-  if (j > 0) {
-    length = j;
+}
+
+struct F {
```
Comment (kelthuzadx) [inline src/hotspot/share/oops/instanceKlass.cpp:1641]: Hi Coleen, is it possible to replace this with existing Pair structure(utilities/pair.hpp)?

Comment (coleenp) [reply src/hotspot/share/oops/instanceKlass.cpp:1641]: Yeah, that would be nicer.  Let me have a look.

Comment (coleenp) [reply src/hotspot/share/oops/instanceKlass.cpp]: Thank you for the improvement.  I was trying to be minimal with 'F' but Pair is better.

**Comment by coleenp** [comment]
Thanks Harold!

**Review by kelthuzadx — APPROVED** [review]
Looks good, thanks for doing this.

**Pull request closed without merging**